core/services/fluxmonitorv2/flux_monitor_test.go

```diff
@@ -0,0 +1,1674 @@
+				AvailableFunds: big.NewInt(int64(tc.funds)),
+				PaymentAmount:  big.NewInt(int64(payment)),
+				OracleCount:    uint8(oracleCount),
+			}
+			assert.Equal(t, tc.want, checker.ExportedSufficientFunds(state))
+		})
+	}
+}
+
+func TestFluxMonitor_SufficientPayment(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	var (
+		fluxAggregator = new(mocks.FluxAggregator)
+		logBroadcaster = new(logmocks.Broadcaster)
+		jobORM         = new(jobmocks.ORM)
+		pipelineORM    = new(pipelinemocks.ORM)
+		spec           = NewSpecification()
+		pollTicker     = fluxmonitorv2.NewPollTicker(time.Minute, false)
+	)
+
+	var payment int64 = 10
+	var eq = payment
+	var gt int64 = payment + 1
+	var lt int64 = payment - 1
+
+	tests := []struct {
+		name               string
+		minContractPayment int64
+		minJobPayment      interface{} // nil or int64
+		want               bool
+	}{
+		{"payment above min contract payment, no min job payment", lt, nil, true},
+		{"payment equal to min contract payment, no min job payment", eq, nil, true},
+		{"payment below min contract payment, no min job payment", gt, nil, false},
+
+		{"payment above min contract payment, above min job payment", lt, lt, true},
+		{"payment equal to min contract payment, above min job payment", eq, lt, true},
+		{"payment below min contract payment, above min job payment", gt, lt, false},
+
+		{"payment above min contract payment, equal to min job payment", lt, eq, true},
+		{"payment equal to min contract payment, equal to min job payment", eq, eq, true},
+		{"payment below min contract payment, equal to min job payment", gt, eq, false},
+
+		{"payment above minimum contract payment, below min job payment", lt, gt, false},
+		{"payment equal to minimum contract payment, below min job payment", eq, gt, false},
+		{"payment below minimum contract payment, below min job payment", gt, gt, false},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			var minJobPayment *assets.Link
+
+			if tc.minJobPayment != nil {
+				mjb := assets.Link(*big.NewInt(tc.minJobPayment.(int64)))
+				minJobPayment = &mjb
+			}
+
+			fm, err := fluxmonitorv2.NewFluxMonitor(
+				NewPipelineRun(),
+				fluxmonitorv2.NewStore(store.DB, store, jobORM, pipelineORM),
+				pollTicker,
+				fluxmonitorv2.NewPaymentChecker(assets.NewLink(tc.minContractPayment), minJobPayment),
+				spec.ContractAddress,
+				fluxmonitorv2.NewFluxAggregatorContractSubmitter(fluxAggregator, transmissionAddress),
+				fluxmonitorv2.NewDeviationChecker(spec.Threshold, spec.AbsoluteThreshold),
+				fluxmonitorv2.Flags{},
+				fluxAggregator,
+				logBroadcaster,
+				spec,
+				func() {},
+				big.NewInt(0),
+				big.NewInt(100000000000),
+			)
+			require.NoError(t, err)
+
+			assert.Equal(t, tc.want, fm.ExportedSufficientPayment(big.NewInt(payment)))
+		})
+	}
+}
+
+func TestFluxMonitor_IsFlagLowered(t *testing.T) {
```

Comment:
This now delegates to the `Flags` but I've left this in for sanity checks for now.

Reply:
Removed